frontend/javascripts/oxalis/api/api_v2.ts

```diff
@@ -555,7 +555,7 @@ class DataApi {
    *
    * api.setMapping("segmentation", mapping);
    */
-  setMapping(layerName: string, mapping: Mapping) {
+  setMapping(layerName: string, mapping: Mapping | Record<number, number>) {
```

Comment:
we call this code internally, right? (e.g., in merger mode) then, it might make sense to avoid the copying in lines 567f somehow :thinking: what do you think?

Reply:
I also see think that it would be way cleaner to avoid mutating the mapping that is already active in the store. At the same time, I'm wondering whether there would be a visible bug, because the mutated mapping is set to the store afterwards, anyway. But obviously we should only do such things, when the performance win is interesting to us. For the merger mode, this is not the case, I think. Not sure whether similar thoughts can be had for your front-end-mapping PR.. However, if you feel like performance is good enough now, let's stick to that :)

Reply:
In the frontend mapping PR, the api functionality is not used. Instead the setMappingAction is dispatched directly. I guess one could do the same in the merger mode too, but as you said, it doesn't seem performance critical for typical merger mode scenarios.